crates/subspace-networking/src/peer_info.rs

```diff
@@ -100,11 +144,23 @@ pub struct Event {
 impl<PIP: PeerInfoProvider> Behaviour<PIP> {
     /// Creates a new `Peer Info` network behaviour with the given configuration.
     pub fn new(config: Config, peer_info_provider: PIP) -> Self {
+        let should_notify_handlers = Arc::new(AtomicBool::new(false));
+        let notify_handler_id = peer_info_provider.subscribe({
+            let should_notify_handlers = should_notify_handlers.clone();
+
+            Arc::new(move |_| {
+                should_notify_handlers.store(true, Ordering::SeqCst);
```

Comment:
This seems expensive. It would have been MUCH nicer if we sent updates rather than the while filter on every change. Imagine the mess that will happen on the network as any farmers do plotting.

Reply:
We specifically chose cuckoo filter because it supports updates. Just like filter can be changed locally, it should be possible to change it remotely in incremental fashion, right?

> The alternative would be sending it after we finish the whole plotting (initial suggestion).

It does not fix the problem with replotting, which will happen everywhere all the time and even a single sector update will result in large complete filter being sent.

Reply:
Half the plot will expire every time history size doubles, so quite often, especially for big plots, especially early during chain history. We need this optimization, we can't just send full filter. If we did and it was infrequent, we'd probably send bloom filters then as they are simpler.

Reply:
You have a point. However, our current design doesn't reflect that and I don't have enough information to address the issue. We need to discuss this separately. Here is a dedicated task for it: https://github.com/subspace/subspace/issues/1627